### ISSUE TITLE
Yarn pnp nodeLinker does not work

### ISSUE THREAD

**Issue by yordis** [sheerlox/import-from-esm#80]
I am using `pnp` linker as follow:

- Node Version: `v20.10.0`

```yaml
# .yarnrc.yml
compressionLevel: mixed
defaultSemverRangePrefix: ''
enableGlobalCache: false
nodeLinker: pnp
yarnPath: .yarn/releases/yarn-4.0.2.cjs
```

I am using `import-from-esm` at https://github.com/straw-hat-team/nodejs-monorepo/tree/master/packages/%40straw-hat/openapi-web-sdk-generator

Here is the config:

```yaml
# .openapi-web-sdk-generatorrc.yaml
generators:
  - path: '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js'
    config:
      outputDir: 'src/lib/services/flashbot-api/operations'
  - path: '@straw-hat/openapi-web-sdk-generator/dist/generators/react-query-fetcher/index.js'
    config:
      outputDir: 'src/lib/services/flashbot-api/react-query'
      packageName: '../operations'

```

And the `package.json`

```json
{
  "packageManager": "yarn@4.0.2",
  "engines": {
    "node": "20.x"
  },
  "type": "module",
  "scripts": {
    "codegen:sdk": "sht-openapi-web-sdk-generator local --config='./openapi.json'"
  }
}
```

When I do `DEBUG=* yarn codegen:sdk`, I get the following outcome:

```
  @straw-hat/openapi-web-sdk-generator:helpers Configuration file /Users/ubi/Developer/myapp/.openapi-web-sdk-generatorrc.yaml loaded +0ms
  @straw-hat/openapi-web-sdk-generator:open-api-web-sdk-generator Loading Generator: @straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js +0ms
  import-from-esm Executing importFrom('/Users/ubi/Developer/myapp', '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js') +0ms
  import-from-esm '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js' is not a file module +0ms
  import-from-esm Trying to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js' from 'file:///Users/ubi/Developer/myapp/noop.js' +0ms
  import-from-esm Failed to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js' from 'file:///Users/ubi/Developer/myapp/noop.js': Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@straw-hat/openapi-web-sdk-generator' imported from /Users/ubi/Developer/myapp/noop.js +1ms
  import-from-esm Trying to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.js' from 'file:///Users/ubi/Developer/myapp/noop.js' +0ms
  import-from-esm Failed to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.js' from 'file:///Users/ubi/Developer/myapp/noop.js': Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@straw-hat/openapi-web-sdk-generator' imported from /Users/ubi/Developer/myapp/noop.js +1ms
  import-from-esm Trying to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.mjs' from 'file:///Users/ubi/Developer/myapp/noop.js' +0ms
  import-from-esm Failed to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.mjs' from 'file:///Users/ubi/Developer/myapp/noop.js': Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@straw-hat/openapi-web-sdk-generator' imported from /Users/ubi/Developer/myapp/noop.js +0ms
  import-from-esm Trying to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.cjs' from 'file:///Users/ubi/Developer/myapp/noop.js' +0ms
  import-from-esm Failed to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.cjs' from 'file:///Users/ubi/Developer/myapp/noop.js': Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@straw-hat/openapi-web-sdk-generator' imported from /Users/ubi/Developer/myapp/noop.js +0ms
  import-from-esm Trying to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.json' from 'file:///Users/ubi/Developer/myapp/noop.js' +0ms
  import-from-esm Failed to resolve '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.json' from 'file:///Users/ubi/Developer/myapp/noop.js': Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@straw-hat/openapi-web-sdk-generator' imported from /Users/ubi/Developer/myapp/noop.js +1ms
  import-from-esm Cannot find module '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js' +0ms
    Error: Failed to load configuration file @straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js.
    Cannot find module '@straw-hat/openapi-web-sdk-generator/dist/generators/fetcher/index.js'
```

If I change the linker to `node-modules` then it works just fine, so something is broken using `pnp` linker and this package.

Any thoughts?!

**Comment by yordis** [comment]
From https://yarnpkg.com/advanced/pnpapi:


Finally, it can be noted that findPnpApi isn't actually needed in most cases and we can do the same with just createRequire thanks to its resolve function:

```js
const {createRequire} = require(`module`);

// We'll be able to inspect the dependencies of the module passed as first argument
const targetModule = process.argv[2];

const targetRequire = createRequire(targetModule);

const resolved = targetRequire.resolve(`eslint`);
const instance = targetRequire(resolved); // <-- still important
```

**Comment by yordis** [comment]
I found the issue, 

```js
function tryResolve(moduleId, baseURL) {
  debug(`Trying to resolve '${moduleId}' from '${baseURL.href}'`);
  try {
    // THIS WORKS
    return import.meta.resolve(moduleId, baseURL.href);
    // THE PROBLEM
    // return moduleResolve(moduleId, baseURL, new Set(['node', 'import']));
  } catch (error) {
    debug(`Failed to resolve '${moduleId}' from '${baseURL.href}': ${String(error)}`);
  }
}
```

**Comment by yordis** [comment]
I created https://github.com/sheerlox/import-from-esm/pull/81 to open the conversation, that solves the issue but I am not sure why it wasn't done that way to begin with, so I am gonna way to find more context about it

**Comment by yordis** [comment]
Related to https://github.com/wooorm/import-meta-resolve/issues/10 

I am a bit confused, but the fact that native `import.meta.resolve` does work but the package does it, I feel the package is at fault here.

https://yarnpkg.com/getting-started/qa#does-yarn-support-esm

**Comment by sheerlox** [comment]
Resolved by #81.

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/import-from-esm/v/1.4.0-alpha.1)
- [GitHub release](https://github.com/sheerlox/import-from-esm/releases/tag/v1.4.0-alpha.1)
